Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -226,6 +227,7 @@ struct PrivateModeStrings {
 class TabTrayController: UIViewController {
     let tabManager: TabManager
     let profile: Profile
+    weak var browserViewController: BrowserViewController?
```

Comment:
Red flags are going off here, too. Passing an entire BVC will only make refactoring BVC harder later if we're injecting it everywhere! Best to either pass an interface that BVC implements or redesign this using delegates, I think.